packages/angular_devkit/build_angular/src/builders/app-shell/render-worker.ts

```diff
@@ -43,29 +60,44 @@ interface RenderRequest {
  * @returns A promise that resolves to the render HTML document for the application.
  */
 async function render({ serverBundlePath, document, url }: RenderRequest): Promise<string> {
-  const { AppServerModule, renderModule, ɵSERVER_CONTEXT } = (await import(serverBundlePath)) as {
-    renderModule: typeof platformServer.renderModule | undefined;
-    ɵSERVER_CONTEXT: typeof platformServer.ɵSERVER_CONTEXT | undefined;
-    AppServerModule: Type<unknown> | undefined;
-  };
+  const {
+    ɵSERVER_CONTEXT,
+    AppServerModule,
+    renderModule,
+    renderApplication,
+    default: bootstrapAppFn,
+  } = (await import(serverBundlePath)) as ServerBundleExports;
 
-  assert(renderModule, `renderModule was not exported from: ${serverBundlePath}.`);
-  assert(AppServerModule, `AppServerModule was not exported from: ${serverBundlePath}.`);
   assert(ɵSERVER_CONTEXT, `ɵSERVER_CONTEXT was not exported from: ${serverBundlePath}.`);
+  assert(renderModule, `renderModule was not exported from: ${serverBundlePath}.`);
```

Comment:
Is `renderModule` needed if `renderApplication` is present?

Reply:
Technically no, although it’s always exported from our builder.

But, thinking about it probably it best to move the asserts in the if blocks.